### PR TITLE
[FIO toup] sig_handler: prefer atomic_uint instead of atomic<bool>

### PR DESCRIPTION
**Pull request by ricardosalveti**
Atomic<bool> operations are not available on GCC 9.2 when building for
RISC-V, causing a build failure. Switch to atomic_uint as that is known
to be provided by GCC in order to avoid the failure.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>

### CODE REVIEW

**Comment by ricardosalveti** [comment]
This is to unblock v54, until the right solution lands in aktualizr.

**Review by doanac — APPROVED** [review]
LGTM

**Pull request merged**
